client/src/layout/menu.js

```diff
@@ -267,7 +267,7 @@ export function fetchMenu(options = {}) {
             tooltip: _l("Account and saved data"),
             menu: [
                 {
-                    title: `${_l("Logged in as")} ${Galaxy.user.get("email")}`,
+                    title: `${_l("Logged in as")} ${Galaxy.user.get("username")}`,
```

Comment:
```suggestion
                    title: `${_l("Logged in as")} ${Galaxy.user.get("username") ? Galaxy.user.get("username") : Galaxy.user.get("email") }`,
```

not every user has username

Reply:
They don't? I thought that was a mandatory field on registration, and should be automatically filled out by ldap if they're using external auth? It's required to construct e.g. published histories URLs, does that fail if they don't have a username set?

Reply:
Is that possible? (not stating - only asking). A User model is required to have a unique username; if we don't fetch that field for the client, we can (should?) change that, I think?

Reply:
@hexylena in my understanding, username doesn't get assign automatically. UI won't let you share history and will force you to specify your username. 
https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/Sharing/Sharing.vue#L11

I suppose, this check won't hurt anyway.

Reply:
@OlegZharkov oh wow, I'd not seen that before, thanks for being aware of it / letting me know! I'll definitely apply this suggestion, just was waiting for the resolution of our discussion :)